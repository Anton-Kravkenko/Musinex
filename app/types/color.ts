export type colorsType = {
	silver: "#E5E5E5";
	charcoal: "#888888";
	dark: "#000";
	primary: "#32B47D";
	midnight: "#101010";
	twilight: "#1B2021";
	dusk: "#202020";
	white: "#fff";
	crimson: "#DC3F41";
	sunshine: "#FFBE0B";
	transparent: "transparent";
}
export type lineColorType = "#E5E5E5" | "#888888" | "#000" | "#32B47D" | "#101010" | "#1B2021" | "#202020" | "#fff" | "#DC3F41" | "#FFBE0B" | "transparent"
export interface ColorProps {
	color?:  lineColorType
}