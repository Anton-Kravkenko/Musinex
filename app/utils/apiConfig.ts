export const SERVER_URL = 'http://10.0.2.2:7777'

export const getAuthUrl = (string: string) => `/auth${string}`
export const getUsersUrl = (string: string) => `/users${string}`
export const getGenresUrl = (string: string) => `/genre${string}`
export const getArtistUrl = (string: string) => `/artist${string}`
export const getAlbumUrl = (string: string) => `/album${string}`
export const getPlaylistUrl = (string: string) => `/playlist${string}`
export const getSearchUrl = (string: string) => `/search${string}`