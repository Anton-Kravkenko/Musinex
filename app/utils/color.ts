const flattenColorPalette =
	require('tailwindcss/lib/util/flattenColorPalette').default
import { theme } from '../../tailwind.config'

const flattenedColors = flattenColorPalette(theme.colors)



export const color = theme.colors
