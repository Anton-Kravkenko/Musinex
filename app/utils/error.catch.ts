export const errorCatch = (error: any): string => {
	console.log(error)
	const message = error?.response?.data?.message

	return message
		? typeof error.response.data.message === 'object'
			? message[0]
			: message
		: error.message
}
