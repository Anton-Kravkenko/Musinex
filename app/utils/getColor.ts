const flattenColorPalette =
	require('tailwindcss/lib/util/flattenColorPalette').default
import { theme } from '../../tailwind.config'

const flattenedColors = flattenColorPalette(theme.colors)

export function getHexCode(colorName: keyof typeof theme.colors) {
	return flattenedColors[colorName]
}
