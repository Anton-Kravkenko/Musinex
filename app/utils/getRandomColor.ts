export function randomBeautifulColor(saturation: number, lightness: number) {
	const hue = Math.floor(Math.random() * 365)
	return `hsl(${hue},${saturation}%,${lightness}%)`
}
