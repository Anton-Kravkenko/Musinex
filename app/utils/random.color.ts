
 export function generateRandomBeautifulHexColor():	string {
	const beautifulColors = [
"#F18805", "#016FB9", "#EF6F6C",	"#F2C078", "#F2E394", "#D96459", "#8C4646", "#E6B89C",   "#4281A4", "#A4D4B4", "#9DC88D", "#F2EE8D", "#1F2041", "#FFC857", "#119DA4", "#19647E", "#C44536", "#197278",  "#5F0F40","#3D3A4B","#1985A1", "#2B2D42","#5FAD56","#F2C14E", "#F78154", "#4D9078", '#2D3047', "#1B998B", "#7A28CB", "#3A445D", "#247BA0", "#6E8894", "#67597A", "#496F5D", "#5863F8", "#141B41", "#9C528B", "#FF3864", "#613F75", "#5C6D70", "#BB4430", "#2FBF71",  "#548687", "#473144"
	
	];
	
	const randomIndex = Math.floor(Math.random() * beautifulColors.length)
	return beautifulColors[randomIndex]
}