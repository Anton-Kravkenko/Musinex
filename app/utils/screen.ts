import { Dimensions } from 'react-native'

export const WindowWidth = Dimensions.get('window').width
export const WindowHeight = Dimensions.get('window').height
export const ScreenWidth = Dimensions.get('screen').width
export const ScreenHeight = Dimensions.get('screen').height