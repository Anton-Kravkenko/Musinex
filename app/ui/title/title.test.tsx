import Title from '@/ui/title/title'
import { Color } from '@/utils/color'
import { cleanup, render, screen, waitFor } from '@testing-library/react-native'

afterEach(() => cleanup())
describe('title', () => {
	it('Render', async function () {
		const titleRender = render(
			<Title
				style={{
					backgroundColor: 'red'
				}}
				className='mb-4'>
				by
			</Title>
		)
		await waitFor(() => expect(screen.queryByTestId('skeleton')).toBeNull())
		expect(titleRender).toMatchSnapshot()
	})
	it('Check Text', async function () {
		const { getByTestId, queryByTestId } = render(<Title>by</Title>)
		await waitFor(() => expect(queryByTestId('skeleton')).toBeNull())
		expect(getByTestId('title')).toHaveTextContent('by')
	})
	it('Check NumberOfLines', async function () {
		const { getByTestId, queryByTestId } = render(
			<Title numberOfLines={2}>by</Title>
		)
		await waitFor(() => expect(queryByTestId('skeleton')).toBeNull())
		expect(getByTestId('title')).toHaveProp('numberOfLines', 2)
	})
	it('Check Color', async function () {
		const { getByTestId, queryByTestId } = render(
			<Title color={Color.crimson}>by</Title>
		)
		await waitFor(() => expect(queryByTestId('skeleton')).toBeNull())
		expect(getByTestId('title')).toHaveProp('color', Color.crimson)
	})
})
