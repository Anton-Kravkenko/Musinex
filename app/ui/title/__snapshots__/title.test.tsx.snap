// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title Render 1`] = `
<Text
  className="mb-4"
  numberOfLines={1}
  style={
    [
      {
        "color": "#fff",
        "fontFamily": "Montserrat_300Light",
        "fontSize": 20,
        "textAlign": "left",
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
  testID="title"
>
  by
</Text>
`;
